src/SymbolicIndexingInterface.jl

```diff
@@ -1,13 +1,104 @@
 module SymbolicIndexingInterface
 
-using DocStringExtensions
+export Symbolic, NotSymbolic, issymbolic, isvariable, variableindex, isparameter,
+    parameterindex, isindependent_variable, isobserved, observed, istimedependent,
+    constant_structure
 
-include("interface.jl")
-include("symbolcache.jl")
+abstract type IsSymbolicTrait end
 
-export independent_variables, is_indep_sym, states, state_sym_to_index, is_state_sym,
-       parameters, param_sym_to_index, is_param_sym, observed, observed_sym_to_index,
-       is_observed_sym, get_state_dependencies, get_observed_dependencies,
-       get_deps_of_observed, SymbolCache, unknown_states
+"""
+    struct Symbolic <: IsSymbolicTrait end
+
+Trait indicating a type is symbolic.
+
+See also: [`NotSymbolic`](@ref), [`issymbolic`](@ref)
+"""
+struct Symbolic <: IsSymbolicTrait end
+
+"""
+    struct NotSymbolic <: IsSymbolicTrait end
+
+Trait indicating a type is not symbolic.
+
+See also: [`Symbolic`](@ref), [`issymbolic`](@ref)
+"""
+struct NotSymbolic <: IsSymbolicTrait end
+
+"""
+    issymbolic(x) = issymbolic(typeof(x))
+    issymbolic(::Type)
+
+Check whether a type implements the [`Symbolic`](@ref) trait or not. Default to
+[`NotSymbolic`](@ref) for all types except `Symbol`.
+"""
+issymbolic(x) = issymbolic(typeof(x))
+issymbolic(::Type) = NotSymbolic()
+issymbolic(::Type{Symbol}) = Symbolic()
```

Comment:
Is there more possibilities than `true/false` from `issymbolic`?

Reply:
I was considering adding a `SymbolicArray` trait variant, for `SymArr` and `AbstractArray{<:something that implements Symbolic}` but it's just as easy to check `all(isequal(Symbolic()), issymbolic.(collect(sym)))`